### PR TITLE
Add `User` and  already generated `IdentityFile` to ssh_config for `nix.linux-builder`

### PR DESCRIPTION
**Pull request by thanegill**

### CODE REVIEW

File: modules/nix/linux-builder.nix [hunk 1]
```diff
@@ -192,9 +192,11 @@ in
 
     environment.etc."ssh/ssh_config.d/100-linux-builder.conf".text = ''
       Host linux-builder
+        User builder
         Hostname localhost
         HostKeyAlias linux-builder
         Port 31022
+        IdentityFile ${cfg.workingDirectory}/keys/builder_ed25519
```
Comment (Enzime) [inline modules/nix/linux-builder.nix:199]:
```suggestion
        IdentityFile /etc/nix/builder_ed25519
```

**Review by Enzime — CHANGES_REQUESTED** [review]

**Review by Enzime — APPROVED** [review]
LGTM

**Comment by Enzime** [comment]
Just wondering, what prompted you to make these changes?

**Comment by thanegill** [comment]
> Just wondering, what prompted you to make these changes?

I have a local nix-cache that I wanted the builder to use. SSH was needed this to debug the connection, found out that the issue was the by default DNS on the linux-builder is [hard-coded to `8.8.8.8`](https://github.com/NixOS/nixpkgs/blob/be6ca51c35ce13ea1b8365aa95946f3643b58db2/nixos/modules/profiles/macos-builder.nix#L104).

**Comment by Montmorency** [comment]
I ran into similar [issue](https://github.com/LnL7/nix-darwin/issues/913) and manually configured the ssh alias to get it working. Thanks for putting this in upstream 👍

**Pull request merged**